### PR TITLE
Fix update, make column nullable

### LINKED ISSUE #13260: Unable to login in some instances upgraded to beta3
After upgrading demo and demo2 to the beta3 (this update applied both beta2 and beta3 SQL upgrades), logging in does not work anymore: although there is no error and it appears to work, I'm still logged out after logging in.

However it seems to work fine on the localhost running beta3 also upgraded from beta1. It will be great to investigate this on demo/demo2 so we can address the root cause.

### PR DESCRIPTION
**Pull request by diosmosis**
Fixes #13260 

CC @mattab

### CODE REVIEW

File: core/Updates/3.6.0-b4.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    public function __construct(MigrationFactory $factory)
+    {
+        $this->migration = $factory;
+    }
+
+    public function getMigrations(Updater $updater)
+    {
+        // use php date since mysql date/time might be different and might lock users out for a while. and subtract days just to be safe.
+        $passwordModified = Date::factory('now')->subDay(14);
```
Comment (mattab) [inline core/Updates/3.6.0-b4.php]: the date-time will be 14 days offset from the actual time where password was changed, would it be possible not to have this 14 days offset somehow? As we may display in the future the time when the password was changed in the UI, the value should be accurate

**Review by mattab — CHANGES_REQUESTED** [review]

File: core/Updates/3.6.0-b4.php (hunk 1 above)
Comment (sgiehl) [reply core/Updates/3.6.0-b4.php]: Then we shouldn't set the value on Matomo update at all and leave it blank. Guess the other code can be adjusted to ignore the additional checks if the password has never changed...

Setting the value to any other value, like `now()` or the user creation date might be wrong when displayed in the UI.

Comment (diosmosis) [reply core/Updates/3.6.0-b4.php]: Makes sense, will make the column null-able

**Comment by diosmosis** [comment]
@mattab modified the PR & tested on demo2

**Pull request merged**